src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,70 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int DEFAULT_CAPACITY = 10;
+    private Pair[] storageOfPairs = new Pair[DEFAULT_CAPACITY];
+    private int size;
+
+    private class Pair<K, V> {
+        private K indexKey;
+        private V data;
+
+        private Pair(K indexKey, V data) {
+            this.indexKey = indexKey;
+            this.data = data;
+        }
+
+        private K getIndexKey() {
+            return indexKey;
+        }
+
+        private void setIndexKey(K indexKey) {
+            this.indexKey = indexKey;
+        }
+
+        private V getData() {
+            return data;
+        }
+
+        private void setData(V data) {
+            this.data = data;
+        }
+    }
+
     @Override
     public void put(K key, V value) {
+        Pair<K, V> pair = new Pair<K, V>(key, value);
+        for (int i = 0; i < size; i++) {
+            if (isDublicate(pair, storageOfPairs[i])) {
+                storageOfPairs[i] = pair;
+                return;
+            }
+        }
+        storageOfPairs[size] = pair;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(storageOfPairs[i].getIndexKey(),key)) {
+                return (V) storageOfPairs[i].getData();
```

Comment:
```suggestion
                return storageOfPairs[i].getData();
```
If we parametrize our Pair class, then we don't need to cast values every time;
 `private Pair<K, V>[] storageOfPairs = new Pair[DEFAULT_CAPACITY];`